routes/donationRouter.js

```diff
@@ -1,8 +1,57 @@
-// const express = require('express');
-// const { db } = require('../server/db');
+const express = require('express');
+const { db } = require('../server/db');
 
-// const donationRouter = express.Router();
+const donationRouter = express.Router();
 
-// Your code below (delete this line when committing):
+donationRouter.get('/', async (req, res) => {
+  try {
+    const donations = await db.query(`
+      SELECT * FROM donation_tracking;
+    `);
+    res.status(200).send(donations);
+  } catch (err) {
+    res.status(500).send(err.message);
+  }
+});
 
-// module.exports = donationRouter;
+donationRouter.get('/:id', async (req, res) => {
+  try {
+    const { id } = req.params;
+    const donation = await db.query('SELECT * FROM donation_tracking WHERE donation_id = $(id)', {
+      id,
+    });
+    res.status(200).send(donation);
+  } catch (err) {
+    res.status(500).send(err.message);
+  }
+});
+
+donationRouter.put('/:donationID', async (req, res) => {
+  try {
+    const { donationID } = req.params;
+
+    const { businessID, foodBankDonation, reporter, email } = req.body;
+
+    if (!businessID) throw new Error('Business ID is required.');
+    if (!foodBankDonation) throw new Error('Food Bank Donation is required.');
+    if (!reporter) throw new Error('Reporter is required.');
+    if (!email) throw new Error('Email is required.');
```

Comment:
Not everything is required, below you have optional parameters, but here youre forcing everything required. This is inconsistent. For a PUT request, everything is optional

Reply:
so make sure to remove these 4 lines